hl7_transformer/hl7_transformer/application.py

```diff
@@ -74,18 +74,51 @@ def _process_message(
         msh_segment = hl7_msg.msh
         logger.debug(f"Message ID: {msh_segment.msh_10.value}")
 
+        transformations_applied = []
+
         created_datetime = msh_segment.msh_7.value
 
         transformed_datetime = transform_datetime(created_datetime)
         msh_segment.msh_7.value = transformed_datetime
 
+        transformations_applied.append(
+                f"DateTime transformed from {created_datetime} to {transformed_datetime}"
+            )
+
+        pid_segment = getattr(hl7_msg, "pid", None)
+        if pid_segment is not None:
+            
+            dod_field = getattr(pid_segment, "pid_29", None)
+            original_dod = (
+                getattr(dod_field, "value", dod_field)
+                if hasattr(dod_field, "value")
+                else dod_field
+            )
```

Comment:
I'm not python expert, how about:

```suggestion
        pid_segment = getattr(hl7_msg, "pid", None)
        if pid_segment:
            dod_field = getattr(pid_segment, "pid_29", None)
            original_dod = getattr(dod_field, "value", dod_field)
```